MetaMask/metamask-mobile
# feat(3299): add tracking to network switching and confirmation

## **Description**
Adds tracking to the user requesting network switching as well as his press of the confirmation button.


## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3299

## **Manual testing steps**

1. Go to the in app browser and open have a connected dapp for example to ethereum main net
2. In the dapp select to change network, the modal will come up, thats when the tracking event if fire for user requesting a network change
3. Confirm the network change by pressing the button, this corresponds to the confirm pressed tracked event

## **Screenshots/Recordings**




#### Mixpanel logs for the two events


<img width="1024" alt="Mixpanel logs for the two events" src="https://github.com/user-attachments/assets/6ab51bd5-42a3-4564-94e3-6c8f76999cb8">

#### Network Switch Requested and Modal Shown
| Interaction     | Tracked event        |
|--------------|--------------|
| <img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/f627e1a2-7503-4f56-ae36-6702077ee7d9"> |<img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/d631bccd-f600-439b-9341-fe76ebbba2a1"> |

#### Network Switch Confirm Pressed
| **Interaction**       | **Tracked event**        |
|--------------|--------------|
| <img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/4b6508d0-c1e3-4dd9-8210-5e63d75b65c0"> |<img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/ecfc708c-b7e2-4530-9fe7-11740c293b12"> |






## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
